examples/tv-casting-app/android/App/app/src/main/java/com/chip/casting/app/CommissionerDiscoveryFragment.java

```diff
@@ -8,143 +8,183 @@
-                          "CommissionerResolveListener.onServiceResolved.OnClickListener.onClick called for "
-                              + discoveredNodeData);
-                      callback.handleCommissioningButtonClicked(discoveredNodeData);
-                    }
-                  });
-              new Handler(Looper.getMainLooper())
-                  .post(
-                      () ->
-                          ((LinearLayout) getActivity().findViewById(R.id.castingCommissioners))
-                              .addView(commissionerButton));
+    private static final String TAG = CommissionerDiscoveryFragment.class.getSimpleName();
+    private static final long DISCOVERY_POLL_INTERVAL_MS = 10000;
+    private static final ArrayList<DiscoveredNodeData> commissionablePlayerList = new ArrayList<>();
+    private FailureCallback failureCallback;
+    private SuccessCallback<DiscoveredNodeData> successCallback;
+    private ScheduledFuture poller;
+    private final TvCastingApp tvCastingApp;
+
+    public CommissionerDiscoveryFragment(TvCastingApp tvCastingApp) {
+        this.tvCastingApp = tvCastingApp;
+    }
+
+    /**
+     * Use this factory method to create a new instance of this fragment using the provided
+     * parameters.
+     *
+     * @return A new instance of fragment CommissionerDiscoveryFragment.
+     */
+    public static CommissionerDiscoveryFragment newInstance(TvCastingApp tvCastingApp) {
+        return new CommissionerDiscoveryFragment(tvCastingApp);
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(
+            LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_commissioner_discovery, container, false);
             }
-          }
-        };
-
-    FailureCallback failureCallback =
-        new FailureCallback() {
-          @Override
-          public void handle(MatterError matterError) {
-            Log.e(TAG, "Error occurred during video player commissioner discovery: " + matterError);
-          }
-        };
-
-    Button discoverButton = getView().findViewById(R.id.discoverButton);
-    discoverButton.setOnClickListener(
-        new View.OnClickListener() {
-          @Override
-          public void onClick(View v) {
-            Log.d(TAG, "Discovering on button click");
-            tvCastingApp.discoverVideoPlayerCommissioners(
-                DISCOVERY_DURATION_SECS, successCallback, failureCallback);
-          }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        Button manualCommissioningButton = getView().findViewById(R.id.manualCommissioningButton);
+        Callback callback = (Callback) this.getActivity();
```

Comment:
It's a bit sketchy to rely on the type of the host `Activity` here. Since this is an example it's not a big deal, but we should probably leave a comment indicating that you wouldn't want to do this in your real Android app.